src/stdlib/redact.rs

```diff
@@ -220,52 +241,162 @@ impl Filter {
+                        b"SHA-256" => hex_digest::<sha_2::Sha256>,
+                        b"SHA-384" => hex_digest::<sha_2::Sha384>,
+                        b"SHA-512" => hex_digest::<sha_2::Sha512>,
+                        b"SHA-512/224" => hex_digest::<sha_2::Sha512_224>,
+                        b"SHA-512/256" => hex_digest::<sha_2::Sha512_256>,
+                        _ => return Err("invalid sha2 variant"),
+                    }
+                } else {
+                    hex_digest::<sha_2::Sha512_256>
+                };
+                Ok(Redactor::Hash(hash))
+            }
+            b"sha3" => {
+                let hash = if let Some(variant) = obj.get("variant") {
+                    match variant
+                        .as_bytes()
+                        .ok_or("`variant must be a string")?
+                        .as_ref()
+                    {
+                        b"SHA3-224" => hex_digest::<sha_3::Sha3_224>,
+                        b"SHA3-256" => hex_digest::<sha_3::Sha3_256>,
+                        b"SHA3-384" => hex_digest::<sha_3::Sha3_384>,
+                        b"SHA3-512" => hex_digest::<sha_3::Sha3_512>,
+                        _ => return Err("invalid sha2 variant"),
+                    }
+                } else {
+                    hex_digest::<sha_2::Sha512_256>
+                };
+                Ok(Redactor::Hash(hash))
+            }
+            _ => Err("unknown `type` for `redactor`"),
         }
     }
 }
 
-impl FromStr for Redactor {
-    type Err = &'static str;
+impl regex::Replacer for &Redactor {
+    fn replace_append(&mut self, caps: &regex::Captures, dst: &mut String) {
+        self.replace_str(&caps[0], dst);
+    }
 
-    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-        use Redactor::Full;
+    fn no_expansion(&mut self) -> Option<Cow<str>> {
+        match self {
+            Redactor::Full => Some(REDACTED.into()),
+            Redactor::Text(s) => Some(s.into()),
+            _ => None,
+        }
+    }
+}
 
-        match s {
-            "full" => Ok(Full),
-            _ => Err("unknown redactor"),
+impl TryFrom<Value> for Redactor {
+    type Error = &'static str;
+
+    fn try_from(value: Value) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Value::Object(object) => Redactor::from_object(object),
+            Value::Bytes(bytes) => match bytes.as_ref() {
+                b"full" => Ok(Redactor::Full),
+                b"sha2" => Ok(Redactor::Hash(hex_digest::<sha_2::Sha512_256>)),
+                b"sha3" => Ok(Redactor::Hash(hex_digest::<sha_3::Sha3_512>)),
+                _ => Err("unknown name of redactor"),
+            },
+            _ => Err("unknown literal for redactor, must be redactor name or object"),
         }
     }
 }
 
+/// Compute the digest of some bytes as hex string
+fn hex_digest<T: digest::Digest>(value: &[u8]) -> String {
+    hex::encode(T::digest(value))
```

Comment:
would it be better to use base64 encoding? or have another option for the text encoding so the caller can choose between hex and base64?

Reply:
As a potential user of this, what would you find most useful? My instinct is that base64 would be more common. Maybe we could use the same approach as we use for the `filters` argument to this function where additional options could be specified, if needed, for a given redactor like:

```
redact("my id is 123456", filters: [r'\d+'], redactor: {name: "sha256", encoding: "base64"})
```

But where you could also just specify:

```
redact("my id is 123456", filters: [r'\d+'], redactor: "sha256")
```

For the default behavior (say base64 encoding). What would you think?

Reply:
> My instinct is that base64 would be more common

that is my instinct as well.